src/braket/circuits/composite_operators.py

```diff
@@ -72,9 +72,17 @@ class QFT(CompositeOperator):
 
     Args:
         qubit_count (int): Number of target qubits.
+        method (bool): Optional boolean flag for decomposition,
+                         with recursive approach by default (default is True),
+                         or non-recursive approach.
     """
 
-    def __init__(self, qubit_count):
+    def __init__(self, qubit_count, method="default"):
```

Comment:
can you make `method` type based such that it's easier to validate and for clients to know what values are valid. Enums are a good approach to this. You can still allow a string version of the enum but it provides the value of clarity in terms of what input is allowed.